lib/src/test/java/io/ably/lib/test/realtime/RealtimeAuthTest.java

```diff
@@ -861,6 +861,76 @@ public Object getTokenRequest(Auth.TokenParams params) {
         }
     }
 
+
+    /**
+     * Call renewAuth() whilst connecting; verify there's no crash (see https://github.com/ably/ably-java/issues/503)
+     */
+    @Test
+    public void auth_renewAuth_whilst_connecting() {
```

Comment:
This test is a bit hard to read and comprehend, could you please refactor it and extract its steps to separate methods? For example the first few lines could possibly be extracted to a method `createTokenDetails()` :wink:

Reply:
Again, this is exactly the same test that was done on `auth_renew_whilst_connecting` with the difference of new function call